src/app/Components/ArtworkFilter/RecentPriceRanges.tsx

```diff
@@ -0,0 +1,156 @@
+import { DEFAULT_PRICE_RANGE as USER_PREFERRED_DEFAULT_PRICE_RANGE } from "app/Scenes/Search/UserPrefsModel"
+import { GlobalStore, useFeatureFlag } from "app/store/GlobalStore"
+import {
+  MAX_SHOWN_RECENT_PRICE_RANGES,
+  useRecentPriceRanges,
+} from "app/store/RecentPriceRangesModel"
+import { Box, Flex, Join, Pill, Spacer, Text, useSpace } from "palette"
+import { useRef } from "react"
+import { Platform, ScrollView, TouchableOpacity } from "react-native"
+import { parsePriceRange, parsePriceRangeLabel, PriceRange } from "./Filters/helpers"
+
+interface RecentPriceRangeEntity {
+  value: string
+  isCollectorProfileSources: boolean
+}
+
+interface RecentPriceRangesListProps {
+  selectedRange: PriceRange
+  ranges: RecentPriceRangeEntity[]
+  onSelected: (priceRange: RecentPriceRangeEntity) => void
+}
+
+interface RecentPriceRangesProps extends Omit<RecentPriceRangesListProps, "ranges"> {}
+
+export const RecentPriceRanges: React.FC<RecentPriceRangesProps> = ({
+  selectedRange,
+  onSelected,
+}) => {
+  const priceRanges = usePriceRanges()
+  const enableRecentPriceRanges = useFeatureFlag("ARRecentPriceRanges")
+  const space = useSpace()
+  const recentPriceRangeScrollRef = useRef<ScrollView>(null)
+  const isEmptyPriceRanges = priceRanges.length === 0
+  const hasCustomPriceRange = priceRanges.some((range) => !range.isCollectorProfileSources)
+
+  const handleClearRecentPriceRanges = () => {
+    GlobalStore.actions.recentPriceRanges.clearAllPriceRanges()
+
+    if (Platform.OS === "android" && recentPriceRangeScrollRef.current) {
+      recentPriceRangeScrollRef.current.scrollTo({ x: 0, animated: false })
+    }
+  }
+
+  if (!enableRecentPriceRanges) {
+    return null
+  }
+
+  return (
+    <>
+      <Flex mx={2} flexDirection="row" justifyContent="space-between" alignItems="center">
+        <Text variant="sm">Recent price ranges</Text>
+
+        {!isEmptyPriceRanges && !!hasCustomPriceRange && (
+          <TouchableOpacity
```

Comment:
Could you also add a11y role and label on this?
```suggestion
          <TouchableOpacity
              accessibilityRole="button"
              accessibilityLabel="Clear price ranges"
```

Reply:
yep, sure

Reply:
done